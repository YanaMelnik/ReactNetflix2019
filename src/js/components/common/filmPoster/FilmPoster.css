.film__poster {
    max-width: 20vw;
}
