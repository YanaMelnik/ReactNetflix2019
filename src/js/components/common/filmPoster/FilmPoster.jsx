import React, { Component } from 'react';
import PropTypes from 'prop-types';

class FilmPoster extends Component {
  render() {
    return (
      <img src={this.props.imgUrl} alt={this.props.filmName}/>
    );
  }
}

FilmPoster.propTypes = {
  imgUrl: PropTypes.string.isRequired,
  filmName: PropTypes.string.isRequired,

};

export default FilmPoster;
