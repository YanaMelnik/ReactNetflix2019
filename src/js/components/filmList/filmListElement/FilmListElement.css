.film-element__header {
    display: flex;
    justify-content: space-between;
    text-transform: uppercase;
    margin: 10px 10px 0;
}

.film-element__title {

}

.film-element__year {
    padding: 1px 5px;
    border: 1px solid black;
    font-size: 11px;
    border-radius: 3px;
    max-height: 13px;
    min-height: 13px;
}

.film-element__genre {
    margin: 0 10px;
    color: #8d8d8d;
    font-size: 12px;
}
