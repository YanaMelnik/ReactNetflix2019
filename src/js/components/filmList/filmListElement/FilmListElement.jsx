import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FilmPoster from '../../common/filmPoster/FilmPoster';

import './FilmListElement.css';

class FilmListElement extends Component {
  render() {
    return (
      <div>
        <FilmPoster filmName={this.props.filmName} imgUrl={this.props.filmPosterUrl}/>

        <div className='film-element__header'>
          <span className='film-element__title'>{this.props.filmName}</span>
          <span className='film-element__year'>{this.props.yearOfIssue}</span>
        </div>

        <span className='film-element__genre'>{this.props.genre}</span>
      </div>
    );
  }
}

FilmListElement.propTypes = {
  filmName: PropTypes.string.isRequired,
  yearOfIssue: PropTypes.number.isRequired,
  genre: PropTypes.string.isRequired,
  filmPosterUrl: PropTypes.string.isRequired,
};

export default FilmListElement;
