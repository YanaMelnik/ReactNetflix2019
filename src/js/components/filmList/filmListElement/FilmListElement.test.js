import React from 'react';
import { shallow } from 'enzyme';
import FilmListElement from './FilmListElement';
import FilmPoster from '../../common/filmPoster/FilmPoster';

describe('FilmListElement', () => {
    let component;
    beforeEach(() => {
        component = shallow(<FilmListElement
            filmName='testName'
            yearOfIssue={1997}
            genre='testGenre'
            filmPosterUrl='testUrl'/>);
    });
    it('should render component correctly in case when all props defined', () => {
        expect(component).toMatchSnapshot();
    });

    it('should always contain FilmPoster component', () => {
        expect(component.find(FilmPoster)).toHaveLength(1);
    });
});
