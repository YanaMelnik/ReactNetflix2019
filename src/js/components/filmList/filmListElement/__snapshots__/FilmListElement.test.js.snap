// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmListElement should render component correctly in case when all props defined 1`] = `
<div
  className="film-element"
>
  <FilmPoster
    filmName="testName"
    imgUrl="testUrl"
  />
  <div
    className="film-element__header"
  >
    <span
      className="film-element__title"
    >
      testName
    </span>
    <span
      className="film-element__year"
    >
      1997
    </span>
  </div>
  <span
    className="film-element__genre"
  >
    testGenre
  </span>
</div>
`;
